lib/adapter.js

```diff
@@ -1691,9 +1752,9 @@ function Adapter(options) {
                         }
                     }
 
-                    typeof options.objectChange === 'function' && setImmediate(() => options.objectChange(id, obj));
+                    typeof options.objectChange === 'function' && setImmediate(options.objectChange, id, obj);
                     // emit 'objectChange' event instantly
-                    setImmediate(() => this.emit('objectChange', id, obj));
+                    setImmediate(this.emit, 'objectChange', id, obj);
```

Comment:
Has nothing to do with the this context because arrow functions are not changing the this context like classic anonymous functions

Reply:
Yes. But if you remove the arrow Funktion the run context is different. Or?!

Reply:
This change will cause the this context to be lost. Passing the member function around is not the same as passing an arrow function around which calls this.something.

Reply:
Don’t really get the difference regarding the `this` context  between calling the function inside an arrow function, which has no own `this` context and calling the function directly. Do you have an example?

Reply:
```js
// repro.js
class Foo {
	constructor() {
		this.prop = "hallo";
	}

	method() {
		console.log(`this is ${JSON.stringify(this)}`);
		console.log(`this.prop is ${this.prop}`);
	}
}

const foo = new Foo();
setTimeout(() => foo.method(), 500);
setTimeout(foo.method, 1000);
```

```
$ node repro.js

this is {"prop":"hallo"}
this.prop is hallo

this is {"_idleTimeout":1000,"_idlePrev":null,"_idleNext":null,"_idleStart":19,"_repeat":null,"_destroyed":false}
this.prop is undefined
```

Reply:
If you don't use the arrow function, foo.method is called with `this` set to the timeout, not `foo`.

Reply:
Yes, this is clear, however, it makes no difference for the `this.emit` call or do I miss out something?

Reply:
I think it should be fine for emit. 

But I still advocate for doing it with arrow functions consistently. Just in case someone uses this pattern in the future, does it like here for consistency and doesn't think twice about the this context.

Reply:
Changing it back just to not break something, but imo it should work with `emit`.